content/razor-views/reading/templates/index.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "Templates"
+date: 2023-02-09T12:48:24-06:00
+draft: false
+weight: 1
+originalAuthor: Courtney Frey # to be set by page creator
+originalAuthorGitHub: speudusa # to be set by page creator
+reviewer: Kimberly Horan # to be set by the page reviewer
+reviewerGitHub: codinglikeagirl42 # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+Take a look at the homepage for [WebElements](https://www.webelements.com/). The content includes text, images, a navigation bar, a search box, linked menu options at the bottom of the page, and 118 carefully colored boxes with links—one for each element on the periodic table. All of this content is very deliberately arranged and styled.
+
+Imagine your boss tasks you with creating this website. Setting up the HTML tags for the navigation bar would be straightforward, but what about the element boxes? You would need to make 118 similar structures, but with different text, links, and colors. Trying to make the table structure work would be tedious at best, and excruciatingly difficult at worst.
+
+Also, what if a new element gets discovered, or some of the data for the elements changes? Updating the text, colors, layout, etc. means adjusting those items in the HTML. Also, if that information appears in other areas of the website, then you need to modify that code as well.
+
+Whew! Changing the website rapidly becomes problematic, especially since it contains lots of data and consists of multiple pages. This is where templates come in to play. They help automate the tasks required to build and maintain a website.
+
+## Templates are Frameworks
+
+A **template** provides the general structure for a web page. Templates outline for us where different elements get placed on the page. Any page made with a template includes its elements and follows its rules. If we add content to the template or modify it in some way, all pages made from that template will reflect the changes.
+
+Let’s see how using a template makes our lives easier.
+
+### No Template Example
+
+The code below displays a simple list. It defines the location for the heading and each `<li>` element, in addition to a couple of fun links. The CSS file (not shown) specifies the font, text size, colors, etc.
+
+```csharp{linenos=table,hl_lines=[],linenostart=1}
```

Comment:
Leaving it up to you,  should these code blocks be in blue example boxes?  Example: http://localhost:1313/classes/reading/background/

Reply:
i put all code examples in an example box.